plugins/GigPlayer/GigPlayer.cpp

```diff
@@ -460,7 +464,11 @@ void GigInstrument::play( sampleFrame * _working_buffer )
 			// Output the data resampling if needed
 			if( resample == true )
 			{
+#ifndef _MSC_VER
 				sampleFrame convertBuf[frames];
+#else
+				const auto convertBuf = static_cast<sampleFrame*>(_alloca(frames * sizeof(sampleFrame)));
+#endif
```

Comment:
Try to avoid calling `_alloca` within a loop - it can easily lead to a stack overflow.

Reply:
What should we use instead? I don't recall where I got this from but I do remember MSVC did not like `convertBuf` as it's defined above.

Reply:
Lift the variable out of the loop if possible - it should be fine to reuse the same buffer for each iteration. There's another instance in this file where the array size is computed inside the loop - in this case, see if it's possible to compute an upper bound for the size.

Eventually, we should get rid of compiler-specific, non-standard stuff like this and preallocate buffers on the heap. However, that's a fairly significant change that's out of scope here, and needs to be applied elsewhere in the codebase too.

Reply:
There is another simple solution: move the loop body to another function to make sure `_alloca` isn't called in every iteration.

Reply:
I found an even simpler solution for now (It's an ugly hack which should be replaced asap but for the current use case, didn't find anything simpler). 

We can add 
```cpp
#ifdef _MSC_VER
_freea(convertBuf)
#endif
```
Just before closing the loop. That way, it'll deallocate the `convertBuf`and avoid the memory leak. Tbh we should be rewriting this part but it's beyond the scope.